### DISCUSSION TITLE
No more than 12 Serverless Functions can be added to a Deployment on the Hobby plan.

### DISCUSSION THREAD

**Discussion by Mood-al** [vercel/community#150] [Help] [answered]
Hey guys,
I'm using the hobby plan of vercel to deploy my site 
but after a certain point this message started to appear at my vercel console 
![image](https://user-images.githubusercontent.com/63562308/140834076-a2018a18-8767-40bc-9e66-978925835926.png)
i didnt use more than 12 serverless functions in my app 
these are my serverless functions that i've created 
![image](https://user-images.githubusercontent.com/63562308/140834263-be8cbc3f-cda1-46f8-a4be-abc9dda60db8.png)

so what is the issue with my work

**Comment by devdev-dev** [comment, marked as answer]
As far as I understand the SSR functions in your pages/components also count as serverless functions but will be bundled.
https://vercel.com/docs/concepts/limits/overview#serverless-functions-created-per-deployment

**Comment by tomkat-cr** [comment]
Although it doesn't have to do with Node.Js, I had the same experience with another kind of API I'm developing. At first I thought it might be the number of endpoints my Python application has with FastAPI, but according to a comment I read on a forum "Every route you declared in the vercel.json file is a serverless function". This means what I declared in "builds":[{"src": "src/*.py"}]. In my case the number of .py files increased from a few to more than 12, but actually my API is concentrated in a single `index.py` file, in which all endpoints are declared. The solution was to change "builds":[{"src": "src/index.py"}] (without the `*.py` wildcard). In other words, each .py file was taken as a function.

**Reply by JoandersonReis** [reply]
It's work for me. Thanks from Brazil &#127463; &#127479;

**Comment by maximeMD** [comment]
Hey there ! 

I just had the the issue with one of my project which did not have any `/pages/api `routes and only a few dynamic pages. The issue was that I had another folder named `/api` in my root folder, and it seems that Vercel considered all the JS/TS code inside this folder as lambda functions to deploy. I just renamed this folder, et voilà 😉

**Reply by thakurfurqaan** [reply]
This was the problem for me.
